signer/src/transaction_signer.rs

```diff
@@ -499,15 +499,52 @@ where
                     return Ok(());
                 }
 
-                let dkg_shares = self
-                    .context
-                    .get_storage()
-                    .get_latest_encrypted_dkg_shares()
-                    .await?;
-
-                if dkg_shares.is_some() {
-                    tracing::warn!("we do not support running DKG more than once");
-                    return Err(Error::DkgHasAlreadyRun);
+                let storage = self.context.get_storage();
+                let config = self.context.config();
+
+                // Get the bitcoin block at the chain tip so that we know the height
+                let bitcoin_chain_tip_block = storage
+                    .get_bitcoin_block(bitcoin_chain_tip)
+                    .await?
+                    .ok_or(Error::NoChainTip)?;
+
+                // Get the number of DKG shares that have been stored
+                let dkg_share_entry_count = storage.get_encrypted_dkg_share_count().await?;
+
+                // Get DKG configuration parameters
+                let dkg_rerun_min_block_height = config.signer.dkg_rerun_bitcoin_height;
+                let dkg_rounds_target = config.signer.dkg_rounds_target;
+
+                // Determine the action based on the DKG share count and the rerun height (if configured)
+                match (
+                    dkg_share_entry_count,
+                    dkg_rounds_target,
+                    dkg_rerun_min_block_height,
+                ) {
+                    (0, _, _) => {
+                        tracing::info!("no DKG shares exist; proceeding with DKG");
+                    }
+                    (current, target, Some(dkg_rerun_height)) => {
+                        if current >= target.get() {
+                            tracing::warn!(
+                                "The target number of DKG shares has been reached; aborting"
+                            );
```

Comment:
nit: adding `current` and `target` in the log could help in debugging (eg, someone forgot to update their confing); also in the `warn` right below.

Reply:
Addressed in [68c948d](https://github.com/stacks-network/sbtc/pull/1205/commits/68c948d30d76e2f33652751eab42a2b80d50c0cb) + [a1f93f4](https://github.com/stacks-network/sbtc/pull/1205/commits/a1f93f4a4497213cd225c1106e18126170ef252e)